tools/js-sdk-release-tools/src/utils/generateInputUtils.ts

```diff
@@ -4,38 +4,49 @@ import { loadTspConfig } from "../common/utils";
 import { RunningEnvironment } from "./runningEnvironment";
 import { exists } from "fs-extra";
 
-async function isManagementPlaneModularClient(specFolder: string, typespecProjectFolder: string[] | string | undefined) {
-    if (!typespecProjectFolder) {
-        return false;
+export async function getSDKType(
+    specFolder: string,
+    readme: string | undefined,
+    typespecProjectFolder: string | undefined
+): Promise<SDKType> {
+    if (readme) {
+        // Update?
+        if (readme.includes("resource-manager")) return SDKType.HighLevelClient;
+        return SDKType.RestLevelClient;
     }
-    
-    if (Array.isArray(typespecProjectFolder) && (typespecProjectFolder as string[]).length !== 1) {
-        throw new Error(`Unexpected typespecProjectFolder length: ${(typespecProjectFolder as string[]).length} (expect 1)`);
+
+    if (!typespecProjectFolder) {
+        const rootCause = `Failed to get SDK Type due to neither README or typespecProjectFolder is undefined.`;
+        throw new Error(rootCause);
     }
 
-    const resolvedRelativeTspFolder = Array.isArray(typespecProjectFolder) ? typespecProjectFolder[0] : typespecProjectFolder as string;
-    const tspFolderFromSpecRoot = path.join(specFolder, resolvedRelativeTspFolder);
-    const tspConfigPath = path.join(tspFolderFromSpecRoot, 'tspconfig.yaml');
+    const tspFolderFromSpecRoot = path.join(specFolder, typespecProjectFolder);
+
+    const tspConfigPath = path.join(tspFolderFromSpecRoot, "tspconfig.yaml");
     if (!(await exists(tspConfigPath))) {
-        return false;
+        const rootCause = `Failed to get SDK Type due to tspconfig.yaml doesn't exist in ${tspFolderFromSpecRoot}.`;
+        throw new Error(rootCause);
     }
 
     const tspConfig = await loadTspConfig(tspFolderFromSpecRoot);
-    if (tspConfig?.options?.['@azure-tools/typespec-ts']?.['isModularLibrary'] !== true) {
-        return false;
-    }
-    return true;
-}
+    const isModularLibrary =
+        tspConfig?.options?.["@azure-tools/typespec-ts"]?.["isModularLibrary"];
 
-// TODO: consider add stricter rules for RLC in when update SDK automation for RLC
-function getSDKType(isMgmtWithHLC: boolean, isMgmtWithModular: boolean) {    
-    if (isMgmtWithHLC) {
-        return SDKType.HighLevelClient;
-    }
-    if (isMgmtWithModular) {
-        return SDKType.ModularClient;
-    }
-    return SDKType.RestLevelClient;
+    // NOTE: respect customer's choice
+    if (isModularLibrary === true) return SDKType.ModularClient;
+    if (isModularLibrary === false) return SDKType.RestLevelClient;
+
+    const isAzureSDK =
+        tspConfig?.options?.["@azure-tools/typespec-ts"]?.["flavor"] ===
+        "azure";
+    // NOTE: unbranded sdk will generate modular client bt default
+    if (!isAzureSDK) return SDKType.ModularClient;
+
+    const isManagementPlane = new RegExp(/\.Management[\/\\]?$/).test(
```

Comment:
I was wondering if we still need this, we already adopted default behavior in codegen side: https://github.com/Azure/autorest.typescript/pull/2973.

Reply:
If so, we don't need to check it.
However, we have an independent changelog generation tool. `GetSDKType()` will be used to bump version - need to find api-version type (stable or preview) based on different client type. I'm wondering if codegen side can add 'api-version' into package.json or somewhere else, then I don't need the check of 'management' any more.

Reply:
Could we directly refer the package name: https://github.com/Azure/azure-rest-api-specs/blob/main/specification/contosowidgetmanager/Contoso.Management/tspconfig.yaml#L39?

Starting with `@azure` would be modular and starting with `@azure-rest` would be RLC.

> I'm wondering if codegen side can add 'api-version' into package.json or somewhere else

I may not get your point, the `api-version` is the package version or spec version?